common/client/types.go

```diff
@@ -16,14 +15,14 @@ import (
 //go:generate mockery --quiet --name RPCClient --output ./mocks --case=underscore
 type RPCClient[
 	CHAIN_ID types.ID,
-	HEAD *evmtypes.Head,
+	HEAD Head,
 ] interface {
 	// ChainID - fetches ChainID from the RPC to verify that it matches config
 	ChainID(ctx context.Context) (CHAIN_ID, error)
 	// Dial - prepares the RPC for usage. Can be called on fresh or closed RPC
 	Dial(ctx context.Context) error
 	// SubscribeToHeads - returns channel and subscription for new heads.
-	SubscribeToHeads(ctx context.Context) (<-chan HEAD, types.Subscription, error)
```

Comment:
Ideally it would,  in this case `chain_client` is wrapping this channel with a `ChainIDSubForwarder` to add the chainID so it can't be read only.

To solve this I can modify the forwarder to return its own new channel instead.